plugins/kubernetes/app/controllers/kubernetes/clusters_controller.rb

```diff
@@ -1,79 +1,42 @@
 # frozen_string_literal: true
-class Kubernetes::ClustersController < ApplicationController
+class Kubernetes::ClustersController < ResourceController
   PUBLIC = [:index, :show].freeze
   before_action :authorize_admin!, except: PUBLIC
   before_action :authorize_super_admin!, except: PUBLIC + [:seed_ecr]
 
-  before_action :find_cluster, only: [:show, :edit, :update, :destroy, :seed_ecr]
+  before_action :find_resource, only: [:show, :edit, :update, :destroy, :seed_ecr]
 
   def new
-    config_filepath = params.dig(:kubernetes_cluster, :config_filepath) || new_config_file_path
-    @cluster = ::Kubernetes::Cluster.new(config_filepath: config_filepath)
-    render :edit
-  end
-
-  def create
-    @cluster = ::Kubernetes::Cluster.new(new_cluster_params)
-    if @cluster.save
-      redirect_to @cluster, notice: "Saved!"
-    else
-      render :edit
-    end
+    super
+    @kubernetes_cluster.config_filepath ||= new_config_filepath
   end
 
   def index
-    @clusters = ::Kubernetes::Cluster.all.sort_by { |c| Samson::NaturalOrder.convert(c.name) }
+    @kubernetes_clusters = ::Kubernetes::Cluster.all.sort_by { |c| Samson::NaturalOrder.convert(c.name) }
     if params[:capacity]
-      @cluster_nodes = Samson::Parallelizer.map(@clusters) do |cluster|
+      @cluster_nodes = Samson::Parallelizer.map(@kubernetes_clusters) do |cluster|
         [cluster.id, cluster.schedulable_nodes]
       end.to_h
     end
   end
 
-  def show
-  end
-
-  def edit
-  end
-
-  def update
-    @cluster.assign_attributes(new_cluster_params)
-    if @cluster.save
-      redirect_to({action: :index}, notice: "Saved!")
-    else
-      render :edit
-    end
-  end
-
-  def destroy
-    if @cluster.destroy
-      redirect_to({action: :index}, notice: "Deleted!")
-    else
-      render :edit
-    end
-  end
-
   def seed_ecr
     SamsonAwsEcr::Engine.refresh_credentials
-    @cluster.namespaces.each do |namespace|
+    @kubernetes_cluster.namespaces.each do |namespace|
       update_secret namespace
     end
     redirect_to({action: :index}, notice: "Seeded!")
   end
 
   private
 
-  def find_cluster
-    @cluster = ::Kubernetes::Cluster.find(params[:id])
-  end
-
-  def new_cluster_params
-    params.require(:kubernetes_cluster).permit(
+  def resource_params
+    super.permit(
       :name, :config_filepath, :config_context, :description, :ip_prefix, deploy_group_ids: []
     )
   end
 
-  def new_config_file_path
+  def new_config_filepath
```

Comment:
Interesting to remove the "_" in file path... okay

Reply:
that was to make all the file+path methods have the same name :)